// @mysten/zksend/dist/cjs/links/zk-bag.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var __accessCheck = (obj, member, msg) => {
  if (!member.has(obj))
    throw TypeError("Cannot " + msg);
};
var __privateGet = (obj, member, getter) => {
  __accessCheck(obj, member, "read from private field");
  return getter ? getter.call(obj) : member.get(obj);
};
var __privateAdd = (obj, member, value) => {
  if (member.has(obj))
    throw TypeError("Cannot add the same private member more than once");
  member instanceof WeakSet ? member.add(obj) : member.set(obj, value);
};
var __privateSet = (obj, member, value, setter) => {
  __accessCheck(obj, member, "write to private field");
  setter ? setter.call(obj, value) : member.set(obj, value);
  return value;
};
var zk_bag_exports = {};
__export(zk_bag_exports, {
  MAINNET_CONTRACT_IDS: () => MAINNET_CONTRACT_IDS,
  ZkBag: () => ZkBag
});
module.exports = __toCommonJS(zk_bag_exports);
var _package, _module;
const MAINNET_CONTRACT_IDS = {
  packageId: "0x5bb7d0bb3240011336ca9015f553b2646302a4f05f821160344e9ec5a988f740",
  bagStoreId: "0x65b215a3f2a951c94313a89c43f0adbd2fd9ea78a0badf81e27d1c9868a8b6fe",
  bagStoreTableId: "0x616db54ca564660cd58e36a4548be68b289371ef2611485c62c374a60960084e"
};
class ZkBag {
  constructor(packageAddress, ids) {
    __privateAdd(this, _package, void 0);
    __privateAdd(this, _module, "zk_bag");
    __privateSet(this, _package, packageAddress);
    this.ids = ids;
  }
  new(tx, {
    arguments: [store, receiver]
  }) {
    tx.moveCall({
      target: `${__privateGet(this, _package)}::${__privateGet(this, _module)}::new`,
      arguments: [
        tx.object(store),
        typeof receiver === "string" ? tx.pure.address(receiver) : receiver
      ]
    });
  }
  add(tx, {
    arguments: [store, receiver, item],
    typeArguments
  }) {
    return tx.moveCall({
      target: `${__privateGet(this, _package)}::${__privateGet(this, _module)}::add`,
      arguments: [
        tx.object(store),
        typeof receiver === "string" ? tx.pure.address(receiver) : receiver,
        tx.object(item)
      ],
      typeArguments
    });
  }
  init_claim(tx, {
    arguments: [store]
  }) {
    const [bag, claimProof] = tx.moveCall({
      target: `${__privateGet(this, _package)}::${__privateGet(this, _module)}::init_claim`,
      arguments: [tx.object(store)]
    });
    return [bag, claimProof];
  }
  reclaim(tx, {
    arguments: [store, receiver]
  }) {
    const [bag, claimProof] = tx.moveCall({
      target: `${__privateGet(this, _package)}::${__privateGet(this, _module)}::reclaim`,
      arguments: [
        tx.object(store),
        typeof receiver === "string" ? tx.pure.address(receiver) : receiver
      ]
    });
    return [bag, claimProof];
  }
  claim(tx, {
    arguments: [bag, claim, id],
    typeArguments
  }) {
    return tx.moveCall({
      target: `${__privateGet(this, _package)}::${__privateGet(this, _module)}::claim`,
      arguments: [tx.object(bag), tx.object(claim), typeof id === "string" ? tx.object(id) : id],
      typeArguments
    });
  }
  finalize(tx, {
    arguments: [bag, claim]
  }) {
    tx.moveCall({
      target: `${__privateGet(this, _package)}::${__privateGet(this, _module)}::finalize`,
      arguments: [tx.object(bag), tx.object(claim)]
    });
  }
  update_receiver(tx, {
    arguments: [bag, from, to]
  }) {
    tx.moveCall({
      target: `${__privateGet(this, _package)}::${__privateGet(this, _module)}::update_receiver`,
      arguments: [
        tx.object(bag),
        typeof from === "string" ? tx.pure.address(from) : from,
        typeof to === "string" ? tx.pure.address(to) : to
      ]
    });
  }
}
_package = new WeakMap();
_module = new WeakMap();
